src/java.base/share/classes/java/lang/invoke/MethodHandleProxies.java

```diff
@@ -373,46 +373,43 @@ private static byte[] createTemplate(ClassLoader loader, ClassDesc proxyDesc, Cl
                                          String methodName, List<MethodInfo> methods) {
         return ClassFile.of(ClassHierarchyResolverOption.of(ClassHierarchyResolver.ofClassLoading(loader)))
                         .build(proxyDesc, clb -> {
-            clb.withSuperclass(CD_Object);
-            clb.withFlags(ACC_FINAL | ACC_SYNTHETIC);
-            clb.withInterfaceSymbols(ifaceDesc);
-
-            // static and instance fields
-            clb.withField(TYPE_NAME, CD_Class, ACC_PRIVATE | ACC_STATIC | ACC_FINAL);
-            clb.withField(TARGET_NAME, CD_MethodHandle, ACC_PRIVATE | ACC_FINAL);
+            clb.withSuperclass(CD_Object)
+               .withFlags(ACC_FINAL | ACC_SYNTHETIC)
+               .withInterfaceSymbols(ifaceDesc)
+               // static and instance fields
+               .withField(TYPE_NAME, CD_Class, ACC_PRIVATE | ACC_STATIC | ACC_FINAL)
+               .withField(TARGET_NAME, CD_MethodHandle, ACC_PRIVATE | ACC_FINAL);
             for (var mi : methods) {
                 clb.withField(mi.fieldName, CD_MethodHandle, ACC_PRIVATE | ACC_FINAL);
             }
 
             // <clinit>
             clb.withMethodBody(CLASS_INIT_NAME, MTD_void, ACC_STATIC, cob -> {
-                cob.loadConstant(ifaceDesc);
-                cob.putstatic(proxyDesc, TYPE_NAME, CD_Class);
-                cob.return_();
+                cob.loadConstant(ifaceDesc)
+                   .putstatic(proxyDesc, TYPE_NAME, CD_Class)
+                   .return_();
             });
 
             // <init>(Lookup, MethodHandle target, MethodHandle callerBoundTarget)
             clb.withMethodBody(INIT_NAME, MTD_void_Lookup_MethodHandle_MethodHandle, 0, cob -> {
-                cob.aload(0);
-                cob.invokespecial(CD_Object, INIT_NAME, MTD_void);
-
-                // call ensureOriginalLookup to verify the given Lookup has access
-                cob.aload(1);
-                cob.invokestatic(proxyDesc, "ensureOriginalLookup", MTD_void_Lookup);
-
-                // this.target = target;
-                cob.aload(0);
-                cob.aload(2);
-                cob.putfield(proxyDesc, TARGET_NAME, CD_MethodHandle);
+                cob.aload(0)
+                   .invokespecial(CD_Object, INIT_NAME, MTD_void)
+                   // call ensureOriginalLookup to verify the given Lookup has access
+                   .aload(1)
+                   .invokestatic(proxyDesc, "ensureOriginalLookup", MTD_void_Lookup)
```

Comment:
Since this class has the `ENSURE_ORIGINAL_LOOKUP` constant, it might be better to use it here:

```suggestion
                   .invokestatic(proxyDesc, ENSURE_ORIGINAL_LOOKUP, MTD_void_Lookup)
```